state-sync/storage-service/server/src/metrics.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use diem_metrics::{
+    register_histogram_vec, register_int_counter_vec, HistogramTimer, HistogramVec, IntCounterVec,
+};
+use diem_types::PeerId;
+use network::ProtocolId;
+use once_cell::sync::Lazy;
+use short_hex_str::AsShortHexStr;
+
+/// Counter for pending network events to the storage service (server-side)
+pub static PENDING_STORAGE_SERVER_NETWORK_EVENTS: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "diem_storage_service_server_pending_network_events",
+        "Counters for pending network events for the storage server",
+        &["state"]
+    )
+    .unwrap()
+});
+
+/// Counter for storage service errors encountered
+pub static STORAGE_ERRORS_ENCOUNTERED: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "diem_storage_service_server_errors",
+        "Counters related to the storage server errors encountered",
+        &["peer", "protocol", "error_type"]
+    )
+    .unwrap()
+});
+
+/// Counter for received storage service requests
+pub static STORAGE_REQUESTS_RECEIVED: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "diem_storage_service_server_requests_received",
+        "Counters related to the storage server requests received",
+        &["peer", "protocol", "request_type"]
```

Comment:
Similarly.  If it's the peer that's the server, remember these dimensions are added by the metrics service.